src/Storages/StorageRedis.cpp

```diff
@@ -0,0 +1,292 @@
+#include "StorageRedis.h"
+#include <Common/parseAddress.h>
+#include <Interpreters/evaluateConstantExpression.h>
+
+#include <Parsers/ASTLiteral.h>
+
+#include <Storages/StorageFactory.h>
+
+#include "Poco/Redis/Redis.h"
+#include "Poco/Redis/Type.h"
+#include "Poco/Redis/Exception.h"
+#include <Poco/Redis/Array.h>
+#include <Poco/Redis/Client.h>
+#include <Poco/Redis/Command.h>
+#include <Processors/Sinks/SinkToStorage.h>
+#include <Common/logger_useful.h>
+#include "Dictionaries/RedisSource.h"
+
+namespace DB
+{
+
+static constexpr size_t REDIS_MAX_BLOCK_SIZE = DEFAULT_BLOCK_SIZE;
+static constexpr size_t REDIS_LOCK_ACQUIRE_TIMEOUT_MS = 5000;
+namespace ErrorCodes
+{
+    extern const int INTERNAL_REDIS_ERROR;
+    extern const int TIMEOUT_EXCEEDED;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+StorageRedis::StorageRedis(
+        const StorageID & table_id_,
+        const std::string & host_,
+        const UInt16 & port_,
+        const UInt32 & db_index_,
+        const std::string & password_,
+        const RedisStorageType & storage_type_,
+        const String & primary_key_column_name_,
+        const String & secondary_key_column_name_,
+        const String & value_column_name_,
```

Comment:
Why do we need those names as separate fields? They are already stored in `ColumnsDescription & columns_`?